test/user_test.go

```diff
@@ -28,63 +27,167 @@ func TestUserBandwidth(t *testing.T) {
+	testUploadSizeSmall := int64(1 + fastrand.Intn(4*database.MiB-1))
+	testUploadSizeBig := int64(4*database.MiB + 1 + fastrand.Intn(4*database.MiB))
+	expectedUploadBandwidth := int64(0)
+	expectedDownloadBandwidth := int64(0)
+
+	// Create a small upload.
 	skylinkSmall, err := createTestUpload(ctx, db, u, testUploadSizeSmall)
 	if err != nil {
 		t.Fatal(err)
 	}
-	details, err := db.UserDetails(ctx, u)
+	expectedUploadBandwidth = database.BandwidthUploadCost(testUploadSizeSmall)
+	// Check the stats.
+	stats, err := db.UserStats(ctx, *u)
 	if err != nil {
-		t.Fatal("Failed to fetch user details.", err)
+		t.Fatal("Failed to fetch user stats.", err)
 	}
-	if details.BandwidthUsed != expectedUploadBandwidthSmall {
-		t.Fatalf("Expected used bandwidth of %d (%d MiB), got %d (%d MiB).", expectedUploadBandwidthSmall, expectedUploadBandwidthSmall/database.MiB, details.BandwidthUsed, details.BandwidthUsed/database.MiB)
+	if stats.NumUploads != 1 {
+		t.Fatalf("Expected a total of %d uploads, got %d.", 1, stats.NumUploads)
+	}
+	if stats.BandwidthUploads != expectedUploadBandwidth {
+		t.Fatalf("Expected upload bandwidth of %d (%d MiB), got %d (%d MiB).",
+			expectedUploadBandwidth, expectedUploadBandwidth/database.MiB,
+			stats.BandwidthUploads, stats.BandwidthUploads/database.MiB)
 	}
 
-	testUploadSizeBig := int64(4*database.MiB + 1 + fastrand.Intn(4*database.MiB))
-	expectedUploadBandwidthBig := database.BandwidthUploadCost(testUploadSizeBig)
+	// Create a big upload.
 	skylinkBig, err := createTestUpload(ctx, db, u, testUploadSizeBig)
 	if err != nil {
 		t.Fatal(err)
 	}
-	details, err = db.UserDetails(ctx, u)
+	expectedUploadBandwidth += database.BandwidthUploadCost(testUploadSizeBig)
+	// Check the stats.
+	stats, err = db.UserStats(ctx, *u)
 	if err != nil {
-		t.Fatal("Failed to fetch user details.", err)
+		t.Fatal("Failed to fetch user stats.", err)
 	}
-	expectedBandwidth := expectedUploadBandwidthSmall + expectedUploadBandwidthBig
-	if details.BandwidthUsed != expectedBandwidth {
-		t.Fatalf("Expected used bandwidth of %d (%d MiB), got %d (%d MiB).", expectedBandwidth, expectedBandwidth/database.MiB, details.BandwidthUsed, details.BandwidthUsed/database.MiB)
+	if stats.NumUploads != 2 {
+		t.Fatalf("Expected a total of %d uploads, got %d.", 2, stats.NumUploads)
+	}
+	if stats.BandwidthUploads != expectedUploadBandwidth {
+		t.Fatalf("Expected upload bandwidth of %d (%d MiB), got %d (%d MiB).",
+			expectedUploadBandwidth, expectedUploadBandwidth/database.MiB,
+			stats.BandwidthUploads, stats.BandwidthUploads/database.MiB)
 	}
 
-	// Register a download.
+	// Register a small download.
 	smallDownload := int64(1 + fastrand.Intn(4*database.MiB))
 	_, err = db.DownloadCreate(ctx, *u, *skylinkSmall, smallDownload)
 	if err != nil {
 		t.Fatal("Failed to download.", err)
 	}
-	// Check bandwidth.
-	details, err = db.UserDetails(ctx, u)
+	expectedDownloadBandwidth += database.BandwidthDownloadCost(smallDownload)
```

Comment:
maybe the `database` package is not the best location for this helper

Reply:
Moved all bandwidth-related costs, vars and functions to a new module called `skynet`.